source/Octopus.Cli/Commands/CommandBase.cs

```diff
@@ -93,10 +93,10 @@ private void PrintJsonHelpOutput(TextWriter writer, string commandName, string d
                         Type = p.Type.Name,
                         Sensitive = p.Sensitive ? (bool?)true : null,
                         AllowsMultiple = p.AllowsMultiple ? (bool?)true : null, //allows tools (such as nuke.build) to auto-generate better code
-                        Values = (p.Type.IsEnum) ? Enum.GetNames(p.Type) : null 
+                        Values = p.Type.IsEnum ? Enum.GetNames(p.Type).Where(x => p.Type.GetField(x).GetCustomAttribute<ObsoleteAttribute>() == null) : null
```

Comment:
I'm missing a bit of context, but should we apply the same logic in `PrintDefaultHelpOutput`?

Reply:
In `PrintDefaultHelpOutput`, we rely on the human provided descriptions. 
We dont automatically print out the valid values